fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/MaterializedViewRewriter.java

```diff
@@ -68,11 +70,44 @@ public OptExpression visitLogicalProject(OptExpression optExpression, Materializ
 
         LogicalProjectOperator projectOperator = (LogicalProjectOperator) optExpression.getOp();
 
+        Map<ColumnRefOperator, ScalarOperator> replaceMap = new HashMap<>();
+        replaceMap.put(context.queryColumnRef, context.mvColumnRef);
+        ReplaceColumnRefRewriter replaceColumnRefRewriter = new ReplaceColumnRefRewriter(replaceMap);
+
         Map<ColumnRefOperator, ScalarOperator> newProjectMap = Maps.newHashMap();
         for (Map.Entry<ColumnRefOperator, ScalarOperator> kv : projectOperator.getColumnRefMap().entrySet()) {
             if (kv.getValue().getUsedColumns().contains(context.queryColumnRef)) {
                 if (kv.getValue() instanceof ColumnRefOperator) {
                     newProjectMap.put(context.mvColumnRef, context.mvColumnRef);
+                } else if (kv.getValue() instanceof CallOperator && kv.getValue().equals(context.conditionCall)) {
+                    Type[] newArgsType = new Type[kv.getValue().getChildren().size()];
+                    if (kv.getValue() instanceof CaseWhenOperator) {
+                        for (int i = 0; i < kv.getValue().getChildren().size(); i++) {
+                            if (((CaseWhenOperator) kv.getValue()).isThenClause(i) ||
+                                    ((CaseWhenOperator) kv.getValue()).isElseClause(i)) {
+                                if (kv.getValue().getChild(i).equals(context.queryColumnRef) ||
+                                        (kv.getValue().getChild(i).isConstantRef() &&
+                                                ((ConstantOperator) kv.getValue().getChild(i)).isNull())) {
+                                    newArgsType[i] = Type.BITMAP;
+                                }
+                            } else {
+                                newArgsType[i] = kv.getValue().getChild(i).getType();
+                            }
+                        }
+                        CaseWhenOperator caseWhenOperator =
+                                new CaseWhenOperator((CaseWhenOperator) kv.getValue(), kv.getValue().getChildren());
+                        caseWhenOperator.setType(Type.BITMAP);
+                        newProjectMap.put(kv.getKey(), replaceColumnRefRewriter.rewrite(caseWhenOperator));
+                    } else {
```

Comment:
why there is only two cases: CaseWhenOperator or IF?
better to add some comment here

Reply:
`isSupportScalarOperator` only supports `case` or `if`.

```
    public static boolean isSupportScalarOperator(ScalarOperator operator) {
        if (operator instanceof CaseWhenOperator) {
            return true;
        }
        return operator instanceof CallOperator &&
                FunctionSet.IF.equalsIgnoreCase(((CallOperator) operator).getFnName());
    }
```